packages/waku/src/lib/plugins/vite-plugin-rsc-index.ts

```diff
@@ -44,19 +44,23 @@ export function rscIndexPlugin(opts: {
     },
     configureServer(server) {
       return () => {
-        server.middlewares.use((req, res) => {
-          server
-            .transformIndexHtml(req.url || '', html)
-            .then((content) => {
-              res.statusCode = 200;
-              res.setHeader('content-type', 'text/html; charset=utf-8');
-              res.end(content);
-            })
-            .catch((err) => {
-              console.error('Error transforming index.html', err);
-              res.statusCode = 500;
-              res.end('Internal Server Error');
-            });
+        server.middlewares.use((req, res, next) => {
+          if (req.url === opts.basePath) {
```

Comment:
the html transform is only needed on the root path to support the "custom" (spa + rsc) mode?

Reply:
This was the major fix. This overrides (continues) Vite's default behavior.